core/toolbox.js

```diff
@@ -511,7 +511,7 @@ Blockly.Toolbox.TreeControl.prototype.enterDocument = function() {
   // Add touch handler.
   if (goog.events.BrowserFeature.TOUCH_ENABLED) {
     var el = this.getElement();
-    Blockly.bindEventWithChecks_(el, goog.events.EventType.TOUCHSTART, this,
+    Blockly.bindEventWithChecks_(el, goog.events.EventType.TOUCHEND, this,
```

Comment:
I'm confused why this changed from TOUCHSTART to TOUCHEND.

Maybe worth taking some of the conversation you had on the other thread and putting it in comments here?

Reply:
Sure.  From @guillaumejenkins on the other thread:

> The following answers are for Chrome, IE & Edge; other browsers are untested
> 
> - Ending a scroll does not select a category
> - Starting a scroll selects the category on Edge & IE
> - Starting a toolbox drag and then moving the touch gesture outside the toolbox is correctly handled (as in, the toolbox correctly scrolls according to your touch gesture even if you dragged outside the toolbox)
> - Ending a drag outside the toolbox does nothing in particular (flyout remains open if it was)
> - It does indeed feel weird that a touch end calls on mouse down... it's been a while and I'm not sure why I changed this from START to END, but it seems like handleTouchEvent_ is never called during a toolbox scroll for some strange reason (tested using a breakpoint), so it might be possible to revert that specific change 
> - The whole idea behind this commit was to let the browser handle the scroll (since the toolbox is a simple div), and to do that I had to remove the call to preventDefault. It does close any context menu as a side effect
>
> Hope this helps clarify

Reply:
The reason `handleTouchEvent_` isn't called during a toolbox scroll is because it was only registered for a touch end.

I switched back to touch start for testing.  The category changes as soon as you touch it, before the scroll starts.  It's able to scroll, but it's very laggy because the flyout takes a while to open.

Switching to touch end means the flyout doesn't open at the beginning of the scroll, and actually it doesn't open at all during a scroll.  It opens if you touch and then release a category, which feels pretty natural--though the function it calls should really be named `onMouseUp` in this case.

I'm looking at whether we want to do the same thing for dragging with a mouse (probably).

Reply:
Updated to touchend and onClick_ to have similar behaviour on mouse or touch.  PTAL.

Reply:
OK. hopefully last question - how's the behavior if you start the click/touch/gesture somewhere outside the toolbox and end it inside?  If that still feels natural, I'm happy.

Reply:
Good check.  It behaves correctly (doesn't open the flyout), probably because my Gesture code is eating the events.